packages/@glimmer/manager/lib/internal/defaults.ts

```diff
@@ -0,0 +1,37 @@
+import { buildCapabilities } from '../util/capabilities';
+
+import type { CapturedArguments as Arguments, HelperCapabilities } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [...Args['positional'], Args['named']]
+  | [...Args['positional'], {}];
+
+interface FunctionHelperState<Args extends Arguments = Arguments> {
+  fn: <Return>(...args: FnArgs<Args>) => Return;
+  args: Args;
+}
+
+export class FunctionHelperManager<State extends FunctionHelperState> {
+  capabilities = buildCapabilities({
+    hasValue: true,
+    hasDestroyable: false,
+    hasScheduledEffect: false,
+  }) as HelperCapabilities;
+
+  createHelper(fn: State['fn'], args: State['args']) {
+    return { fn, args };
+  }
+
+  getValue({ fn, args }: State) {
+    let argsForFn: FnArgs<Arguments> = [
+      ...args.positional,
+      Object.keys(args.named).length > 0 ? args.named : {},
+    ];
```

Comment:
This seems unnecessary – if we got here by definition we don't have arguments so we just need to call the function. However, there is the open question of whether it should be `fn()` vs `fn({})`.

Reply:
I think it should be `fn({})`, because at the call site, it's a common pattern (that I've seen anyway (outside of ember)), to do like:
```js
let options = args[args.length - 1];
// or
let [options] = args.reverse();
```
giving a stable type is good DX, imo, and reduces the amount of surprise when arguments aren't passed.

Reply:
If that's the case then you can't ever write something like this:

```js
class MyComponent {
  @tracked name = '...';

  greet(message = 'hello') { // <- this will always be filled with `{}`
    return `${message}, ${this.name}`;
  }
}
```

More importantly, you can't import and use a lot of library functions that uses patterns like this either (it doesn't have to be the only/first argument – as long as it takes a variable number of positional arguments, i.e. any function with at least one default positional argument – then the result would be wrong.

Which seems like a big deal considering this is supposed to make it possible to just import an existing function and use it in the template directly.

Reply:
yeah, that's true. :thinking: 
default arguments are enough to have me switch the implementation. Next commit

Reply:
Let's make sure we test it too, and try to give this some visibility so people other than us can chime in (a PR to update the RFC is probably the easiest to get attention).

This is just one possible path, but there is also the more generic versions, such as `{{(someFunction)}}`. What happens in that case?

Reply:
> This seems unnecessary – if we got here by definition we don't have arguments so we just need to call the function.

Sorry – reading the code more, I think I am wrong about this (doesn't affect the `()` vs `({})` discussion but we should probably make that its own thread). I thought `getValue()` is only called in the `{{someFn}}` case, but that doesn't seem like how it works and is just the part of the helper manager api that mans "call the helper". So I think your args-handling code is actually needed.